charset/convert.go

```diff
@@ -3,23 +3,27 @@ package charset
 import (
 	"bytes"
 	"io"
+
+	"golang.org/x/text/encoding/charmap"
 )
 
-func ToUtf8(iso8859_1_buf []byte) string {
-	buf := make([]rune, len(iso8859_1_buf))
-	for i, b := range iso8859_1_buf {
-		buf[i] = rune(b)
+func mustConvert(b []byte, err error) []byte {
+	if err != nil {
+		panic(err)
 	}
-	return string(buf)
+	return b
+}
+
+func ToUtf8(iso8859_1_buf []byte) string {
+	decoder := charmap.ISO8859_1.NewDecoder()
+	// TODO: propagate errors
+	return string(mustConvert(decoder.Bytes(iso8859_1_buf)))
 }
 
 func ToISO8859_1(utf8String string) []byte {
```

Comment:
The reason for `_1` is that there are multiple encodings defined in ISO 8859. `_1` is commonly know as Latin1.